### PR TITLE
[Certificates] Add check for certificate date based on creation date of the user

### PR DESCRIPTION
**Pull request by legionth**
Adds a check for the certificate migration. New users (after installing the persisting certificate feature) won't see the migration option.

### CODE REVIEW

**Comment by legionth** [comment]
Ping @mjansenDatabay

File: Services/Certificate/classes/Migration/class.ilCertificateMigrationValidator.php [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+	public function __construct(ilSetting $certificateSettings)
+	{
+		$this->certificateSettings = $certificateSettings;
+	}
+
+	/**
+	 * @param ilObjUser $user
+	 * @return bool
+	 */
+	public function isMigrationAvailable(ilObjUser $user)
```
Comment (mjansenDatabay) [inline Services/Certificate/classes/Migration/class.ilCertificateMigrationValidator.php]: @legionth RTD is missing here

File: Services/Certificate/classes/Migration/class.ilCertificateMigrationValidator.php [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+	{
+		$userCreationDate = $user->getCreateDate();
+
+		$showMigrationBox = false;
+		if (null !== $userCreationDate) {
+			$userCreatedTimestamp = strtotime($userCreationDate);
+			$introducedTimestamp = $this->certificateSettings->get('persisting_cers_introduced_ts');
```
Comment (mjansenDatabay) [inline Services/Certificate/classes/Migration/class.ilCertificateMigrationValidator.php]: @legionth Maybe you should use a meaningful default here as 2nd parameter

**Review by mjansenDatabay — CHANGES_REQUESTED** [review]

File: Services/Certificate/classes/Migration/class.ilCertificateMigrationValidator.php [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+		$this->certificateSettings = $certificateSettings;
+	}
+
+	/**
+	 * @param ilObjUser $user
+	 * @return bool
+	 */
+	public function isMigrationAvailable(ilObjUser $user)
+	{
+		$userCreationDate = $user->getCreateDate();
```
Comment (mjansenDatabay) [inline Services/Certificate/classes/Migration/class.ilCertificateMigrationValidator.php]: @legionth / @qdittrich : Maybe other *preconditions* should be added here as well

**Review by mjansenDatabay — APPROVED** [review]

**Comment by legionth** [comment]
Resolved Merge Conflicts :shipit: .

**Comment by mjansenDatabay** [comment]
@legionth :shipit: not possible due to conflict

**Comment by legionth** [comment]
Fast now @mjansenDatabay :sweat_smile:

**Pull request merged**